zboxcore/zboxutil/http.go

```diff
@@ -354,6 +354,25 @@ func NewListRequest(baseUrl, allocation string, path string, auth_token string)
 	return req, nil
 }
 
+// NewUploadRequestWithMethod create a http reqeust of upload
+func NewUploadRequestWithMethod(baseURL, allocation string, body io.Reader, method string) (*http.Request, error) {
```

Comment:
There should be a unit test to confirm `NewUploadRequestWithMethod` returns the right http.Request. I know no one else has done it, but that's now reason why this PR should make the same mistake.